### PR TITLE
feat: Add GitHub action theme auto label

### PR DESCRIPTION
**Pull request by error7404**
Github action to check if a source request has Madara theme or lighnovelwp theme and automatically add the correct label to it (more multisrc can be added but would need to find unique identifiers)

image

---

will add a comment when the action cannot access the URL that was sent

image

### CODE REVIEW

File: .github/workflows/theme_auto_label.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      - name: Check if label is "Source Request"
+        id: label_check
+        run: |
+          LABEL_NAME=$(jq --raw-output .label.name "$GITHUB_EVENT_PATH")
+          if [[ "$LABEL_NAME" == "Source Request" ]]; then
+            echo "source_request=true" >> $GITHUB_OUTPUT
+          else
+            echo "source_request=false" >> $GITHUB_OUTPUT
+            echo "# ❌ Skipping: label is not Source request" >> $GITHUB_STEP_SUMMARY
+          fi
```
Comment (nyagami) [inline .github/workflows/theme_auto_label.yml:21]:
```suggestion
    if: contains(github.event.issue.labels.*.name, 'Source Request')
    runs-on: ubuntu-latest
    steps:
```
https://docs.github.com/en/actions/learn-github-actions/expressions#example-using-an-object-filter

File: .github/workflows/theme_auto_label.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+      - name: Get Source Request URL
+        if: steps.label_check.outputs.source_request == 'true'
```
Comment (nyagami) [inline .github/workflows/theme_auto_label.yml:24]:
```suggestion
```

File: .github/workflows/theme_auto_label.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        id: get_url
+        run: |
+          ISSUE_BODY=$(jq --raw-output .issue.body "$GITHUB_EVENT_PATH")
```
Comment (nyagami) [inline .github/workflows/theme_auto_label.yml:27]: you can access issue body via context `github.event.issue.body`

File: .github/workflows/theme_auto_label.yml (hunk 1 above)
Comment (error7404) [reply .github/workflows/theme_auto_label.yml:21]: I've read it but my brain never caught it...

File: .github/workflows/theme_auto_label.yml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO: ${{ github.repository }}
+          GITHUB_ISSUE_NUMBER: ${{ steps.get_issue_number.outputs.GITHUB_ISSUE_NUMBER }}
+        run: |
+          if [[ -z "$URL" ]]; then
+            echo "url_valid=false" >> $GITHUB_OUTPUT
+            echo "# ❌ Error: URL not found" >> $GITHUB_STEP_SUMMARY
+            exit 1
+          else
+            if curl --output /dev/null --silent --head --fail "$URL"; then
```
Comment (nyagami) [inline .github/workflows/theme_auto_label.yml:53]: I think adding an user agent in headers could increase accessibility

**Review by nyagami — APPROVED** [review]

**Pull request merged**